applications/IgaApplication/custom_utilities/brep_json_io.cpp

```diff
@@ -0,0 +1,454 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:     BSD License
+//           Kratos default license: kratos/IGAStructuralMechanicsApplication/license.txt
+//
+//  Main authors:    Tobias Teschemacher
+//                   Riccardo Rossi
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "brep_json_io.h"
+#include "iga_application.h"
+#include "iga_application_variables.h"
+
+
+namespace Kratos
+{
+    void BrepJsonIO::WriteIntegrationDomainJson(ModelPart& rModelPart, const std::string& rOutputFileName)
+    {
+        std::ofstream file;
+        file.open(rOutputFileName);
+        std::string separator = "";
+        file << "{\"geometry_integration_points\" :[";
+        for (auto element = rModelPart.ElementsBegin(); element != rModelPart.ElementsEnd(); ++element)
+        {
+            const Vector& local_parameters = element->GetValue(LOCAL_COORDINATES);
+            //const Vector& global_coordinates = element->Calculate(COORDINATES);
+            file << separator << "[" << element->Id() << ", " << element->GetValue(FACE_BREP_ID) << ",[" << local_parameters[0] << ", " << local_parameters[1] << "]]";
+            separator = ",";
+        }
+        file << "]}";
+        file.close();
+    }
+
+    std::vector<BrepModel> BrepJsonIO::ImportNurbsBrepGeometry(
```

Comment:
![2mmfnh](https://user-images.githubusercontent.com/25484702/48555314-2290ca00-e8e1-11e8-802d-bcae7c930569.jpg)

Please refactor this into smaller fcts, this is ~400 lines long!

E.g. into edges, faces etc
Please take into account the comment I made abt other parts of the code